### PR TITLE
#1833: Add ability to disable renditions functionality to stores configuration

### LINKED ISSUE #1833: Add ability to disable renditions functionality to stores configuration
### Steps to reproduce (*)
1. Go to Stores -> Configuration -> Advanced -> System
2. Expand "Media Gallery Renditions" fieldset

### Expected result (*)
"Enable Image Optimization" select with "Yes/No" options is added as the first field of "Media Gallery Image Optimization" fieldset
The value is stored in the `system/media_gallery_renditions/enabled` path and is `1` by default (`app/code/Magento/MediaGalleryRenditions/etc/config.xml`)
If renditions are disabled - renditions should not be generated during insertion of the image and during the width/height change. (add condition to `UpdateRenditionsOnConfigChange` and `SetRenditionPath`  plugins)
When renditions are enabled - all renditions should be regenerated (`\Magento\MediaGalleryRenditions\Plugin\UpdateRenditionsOnConfigChange`)

image

The screenshot includes also the changes done in https://github.com/magento/magento2/pull/30044

### PR DESCRIPTION
**Pull request by joweecaquicla**
Introduction of Media Gallery Renditions functionality is approved in https://jira.corp.magento.com/browse/MC-37383 (SVC failure is approved)

### Description (*)

This PR will introduce the ability to disable renditions functionality to stores configuration and additional conditions to rendition-related classes.
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/adobe-stock-integration/issues/1833

### Manual testing scenarios (*)

Previously Rendition functionality was enabled/disabled based on Enhanced Media Gallery enable/disable value.
Currently  t has a separate switch in stores configuration.

"Enable Image Optimization" select with "Yes/No" options is added as the first field of "Media Gallery Image Optimization" fieldset
"Enable Image Optimization" is enabled by default
If renditions are disabled - renditions should not be generated during insertion of the image and during the width/height change.
When renditions are enabled - all renditions should be regenerated.
Renditions functionality should work for both old and new media gallery.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryRenditions/Model/Config.php [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Config
 {
     private const TABLE_CORE_CONFIG_DATA = 'core_config_data';
     private const XML_PATH_ENABLED = 'system/media_gallery/enabled';
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryRenditions/Model/Config.php]: This constant is no longer needed and can be removed

File: app/code/Magento/MediaGalleryRenditions/Model/Config.php [hunk 2]
```diff
@@ -54,6 +55,16 @@ public function isEnabled(): bool
         return $this->scopeConfig->isSetFlag(self::XML_PATH_ENABLED);
     }
 
+    /**
+     * Check if the media gallery renditions is enabled
+     *
+     * @return bool
+     */
+    public function isMediaGalleryRenditionsEnabled(): bool
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryRenditions/Model/Config.php]: New method introduction is not necessary, `isEnabled` method should be now based on `system/media_gallery_renditions/enabled` value

File: app/code/Magento/MediaGalleryRenditions/Plugin/SetRenditionPath.php [hunk 3]
```diff
@@ -91,7 +91,7 @@ public function beforeExecute(
             $storeId
         ];
 
-        if (!$this->config->isEnabled()) {
+        if (!$this->config->isEnabled() || !$this->config->isMediaGalleryRenditionsEnabled()) {
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryRenditions/Plugin/SetRenditionPath.php]: no change is needed here

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @joweecaquicla ! Please see my  comments

As for the test coverage:
Can you please add an ActionGroup enabling the renditions and use this action group for existing New Media Gallery test suites. I believe that should be enough

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Review by sivaschenko — DISMISSED** [review]

**Comment by sivaschenko** [comment]
Introduction of Media Gallery Renditions  functionality is approved in https://jira.corp.magento.com/browse/MC-37383 (SVC is approved)

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8223](https://jira.corp.magento.com/browse/ENGCOM-8223) has been created to process this Pull Request

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
:x: QA Failed

When the **Enable Image Optimization** is set to **No**, Config is Saved and Cache is cleared
endis_im_optim_no
Still, the rendition images are inserted into newly created Page, Category...
The Expected behaviour is to insert original image sizes.

**Comment by sivaschenko** [comment]
@joweecaquicla looks like the update for `\Magento\MediaGalleryRenditions\Model\Config` is missing.

private const XML_PATH_ENABLED = 'system/media_gallery/enabled'

should be changed to 

private const XML_PATH_ENABLED = 'system/media_gallery_renditions/enabled'

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8223](https://jira.corp.magento.com/browse/ENGCOM-8223) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Enable Image Optimization** option is added under _Stores - Configuration - Advanced - System - Media Gallery Renditions_ 
endis_im_optim

The default value is **Yes**

When the value is set to Yes, the images added to the content are optimized to the size set in _Maximum Width_ and _Maximum Height_ fields

When the value is set to **No**, the original image size is inserted into the Content of an Entity

When **No** and **Yes** values are changed, and `bin/magento queue:consumers:start media.gallery.renditions.update` is performed, the images that have already been inserted with original size, are NOT reduced or enlarged.

Manual testing scenario - https://studio.cucumber.io/projects/131313/test-plan/folders/1507933/scenarios/5219418

**Review by engcom-Bravo — COMMENTED** [review]
:heavy_check_mark: QA Passed

**Enable Image Optimization** option is added under _Stores - Configuration - Advanced - System - Media Gallery Renditions_ 
endis_im_optim

The default value is **Yes**

When the value is set to Yes, the images added to the content are optimized to the size set in _Maximum Width_ and _Maximum Height_ fields

When the value is set to **No**, the original image size is inserted into the Content of an Entity

When **No** and **Yes** values are changed, and `bin/magento queue:consumers:start media.gallery.renditions.update` is performed, the images that have already been inserted with original size, are NOT reduced or enlarged.

Manual testing scenario - https://studio.cucumber.io/projects/131313/test-plan/folders/1507933/scenarios/5219418

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@sivaschenko SVC failures must be approved.

**Comment by sivaschenko** [comment]
Introduction of Media Gallery Renditions functionality is approved in https://jira.corp.magento.com/browse/MC-37383 (SVC failure is approved)

**Pull request merged**